extension/src/openvic-extension/classes/GUINode.cpp

```diff
@@ -207,17 +210,46 @@ APPLY_TO_TEXTURE_TYPES(TEXTURE_GET_FUNCTIONS)
 Error GUINode::hide_node(NodePath const& path) const {
 	CanvasItem* node = _cast_node<CanvasItem>(get_node_internal(path));
 	ERR_FAIL_NULL_V(node, FAILED);
+
 	node->hide();
+
 	return OK;
 }
 
 Error GUINode::hide_nodes(TypedArray<NodePath> const& paths) const {
 	Error ret = OK;
+
 	for (int32_t i = 0; i < paths.size(); ++i) {
 		if (hide_node(paths[i]) != OK) {
 			ret = FAILED;
 		}
 	}
+
+	return ret;
+}
+
+Error GUINode::remove_node(NodePath const& path) const {
+	Node* node = get_node_internal(path);
+	ERR_FAIL_NULL_V(node, FAILED);
+
+	Node* parent = node->get_parent();
+	ERR_FAIL_NULL_V(parent, FAILED);
+
+	parent->remove_child(node);
+	node->queue_free();
```

Comment:
queue_free automatically does this, there is no need to remove a child for a node that calls queue_free